src/payment/strategies/paypal/paypal-script-loader.ts

```diff
@@ -6,22 +6,60 @@ import { PaypalHostWindow, PaypalSDK } from './paypal-sdk';
 
 export default class PaypalScriptLoader {
     private _window: PaypalHostWindow;
+    private _scripts: { [key: string]: Promise<Event> } = {};
 
     constructor(
         private _scriptLoader: ScriptLoader
     ) {
         this._window = window;
     }
 
-    loadPaypal(): Promise<PaypalSDK> {
-        return this._scriptLoader
-            .loadScript('//www.paypalobjects.com/api/checkout.min.js')
-            .then(() => {
-                if (!this._window.paypal) {
-                    throw new PaymentMethodClientUnavailableError();
-                }
+    loadPaypal(merchantId: string = ''): Promise<PaypalSDK> {
+        if (merchantId.length > 0) {
+            return this.loadScriptWithAttribute('//www.paypalobjects.com/api/checkout.min.js', 'data-merchant-id', merchantId)
+                .then(() => {
+                    if (!this._window.paypal) {
+                        throw new PaymentMethodClientUnavailableError();
+                    }
 
-                return this._window.paypal;
+                    return this._window.paypal;
+                });
+        } else {
+            return this._scriptLoader
+                .loadScript('//www.paypalobjects.com/api/checkout.min.js')
+                .then(() => {
+                    if (!this._window.paypal) {
+                        throw new PaymentMethodClientUnavailableError();
+                    }
+
+                    return this._window.paypal;
+                });
+        }
+    }
+
+    loadScriptWithAttribute(src: string, attribute: string, value: string): Promise<Event> {
+        if (!this._scripts[src]) {
```

Comment:
Considering we have full control over the script loader. Do you think it will be feasible to add this functionality directly there? It seems like a bit of duplication doing a whole custom script loader for this.